tensorflow/python/eager/forwardprop.py

```diff
@@ -326,7 +326,7 @@ class ForwardAccumulator(object):
   <tf.Tensor: shape=(2,), dtype=float32, numpy=array([6., 0.], dtype=float32)>
   """
 
-  def __init__(self, primals, tangents):
+  def __init__(self, primals, tangents, use_batch=False):
```

Comment:
Can we keep this out of the public API to start, e.g. a separate static method with this signature (_batch_accumulator)?

Once it's tested with tf.function/custom_gradient/nesting we can start with experimental_use_batch. Just a question of what's guaranteed about API stability.

Reply:
That seems better. So instead of a `self` argument a function? Can we make it a `@classmethod`?

Reply:
Sure, a new _private classmethod sounds reasonable to me for now.